src/main/java/keycontacts/model/Model.java

```diff
@@ -85,4 +87,9 @@ public interface Model {
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredStudentList(Predicate<Student> predicate);
+
+    /**
+     * Returns a {@Code ClashResult} object containing details of lesson clashes.
+     */
+    public ClashResult checkClashingLesson(Lesson lessonToCheck);
```

Comment:
This method should be shifted up, to the section for student directory methods

Reply:
Also, I think the name of the method can be a little clearer, maybe 'getClashResult' instead? Open to discussion